service_container.rst

```diff
@@ -5,92 +5,90 @@
 
-What is a Service?
-------------------
+    use Symfony\Bundle\FrameworkBundle\Controller\Controller;
 
-Put simply, a service is any PHP object that performs some sort of "global"
-task. It's a purposefully-generic name used in computer science to describe an
-object that's created for a specific purpose (e.g. delivering emails). Each
-service is used throughout your application whenever you need the specific
-functionality it provides. You don't have to do anything special to make a
-service: simply write a PHP class with some code that accomplishes a specific
-task. Congratulations, you've just created a service!
+    class ProductController extends Controller
+    {
+        /**
+         * @Route("/products")
+         */
+        public function listAction()
+        {
+            $logger = $this->container->get('logger');
+            $logger->info('Look! I just used a service');
 
-.. note::
+            // ...
+        }
+    }
 
-    As a rule, a PHP object is a service if it is used globally in your
-    application. A single ``Mailer`` service is used globally to send
-    email messages whereas the many ``Message`` objects that it delivers
-    are *not* services. Similarly, a ``Product`` object is not a service,
-    but an object that persists ``Product`` objects to a database *is* a service.
-
-So what's the big deal then? The advantage of thinking about "services" is
-that you begin to think about separating each piece of functionality in your
-application into a series of services. Since each service does just one job,
-you can easily access each service and use its functionality wherever you
-need it. Each service can also be more easily tested and configured since
-it's separated from the other functionality in your application. This idea
-is called `service-oriented architecture`_ and is not unique to Symfony
-or even PHP. Structuring your application around a set of independent service
-classes is a well-known and trusted object-oriented best-practice. These skills
-are key to being a good developer in almost any language.
+``logger`` is a unique key for the ``Logger`` object. What other services are available?
+Find out by running:
 
-.. index::
-   single: Service Container; What is a service container?
+.. code-block:: terminal
 
-What is a Service Container?
-----------------------------
+    php bin/console debug:container
 
-A service container (or *dependency injection container*) is simply a PHP
-object that manages the instantiation of services (i.e. objects).
+This is just a *small* sample of the output:
 
-For example, suppose you have a simple PHP class that delivers email messages.
-Without a service container, you must manually create the object whenever
-you need it::
+Service ID                      Class name
+------------------------------- -------------------------------------------------------------------
+doctrine                        Doctrine\Bundle\DoctrineBundle\Registry
```

Comment:
we have to wrap these values inside literals, otherwise the backslashes won't be rendered